cmd/kube-apiserver/app/aggregator.go

```diff
@@ -146,16 +146,56 @@ func createAggregatorServer(aggregatorConfig *aggregatorapiserver.Config, delega
 }
 
 func makeAPIService(gv schema.GroupVersion) *apiregistration.APIService {
+	apiServicePriority, ok := apiVersionPriorities[gv]
+	if !ok {
+		// if we aren't found, assume we're a CRD
+		apiServicePriority = priority{group: 5000, version: 20}
+	}
 	return &apiregistration.APIService{
 		ObjectMeta: metav1.ObjectMeta{Name: gv.Version + "." + gv.Group},
 		Spec: apiregistration.APIServiceSpec{
-			Group:    gv.Group,
-			Version:  gv.Version,
-			Priority: 100,
+			Group:           gv.Group,
+			Version:         gv.Version,
+			GroupPriority:   apiServicePriority.group,
+			VersionPriority: apiServicePriority.version,
 		},
 	}
 }
 
+type priority struct {
+	group   int32
+	version int32
+}
+
+// The proper way to resolve this problem is to refactor the genericapiserver.DelegationTarget to include a list of priorities based on which APIs were installed.
```

Comment:
letting the aggregator know the desired group and version-within-group order of the underlying servers